src/file/functions.rs

```diff
@@ -393,7 +393,7 @@ fn get_partition_info(image_file: &str, partition: &Partition) -> Result<Partiti
         Partition::boot => 1,
         Partition::rootA => 2,
         p @ (Partition::factory | Partition::cert) => {
-            let re = Regex::new(r"Disklabel type: (\D{3})").unwrap();
+            let re = Regex::new(r": (dos|gpt)").unwrap();
```

Comment:
I suggest to don't use fdisk (we don't know if it is installed, and which version, e.g. busybox and so on).
We should look into crates which read partition tables.